lux-lib/src/operations/sync.rs

```diff
@@ -238,123 +246,122 @@ async fn do_sync(
-            assert!(!report.added.is_empty());
-            assert!(!report
-                .added
-                .iter()
-                .filter(|pkg| pkg.name().to_string() == "toml-edit")
-                .collect_vec()
-                .is_empty());
-        }
-        let lockfile_after_sync = ProjectLockfile::new(lockfile_path).unwrap();
-        assert!(!lockfile_after_sync
-            .rocks(&LocalPackageLockType::Regular)
-            .is_empty());
-    }
-
-    #[tokio::test]
-    async fn test_sync_remove_rocks() {
-        let tree_path =
-            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("resources/test/sample-tree");
-        let temp_dir = TempDir::new().unwrap();
-        temp_dir.copy_from(&tree_path, &["**"]).unwrap();
-        let empty_lockfile_dir = TempDir::new().unwrap();
-        let lockfile_path = empty_lockfile_dir.path().join("lux.lock");
-        let mut empty_lockfile = ProjectLockfile::new(lockfile_path.clone())
-            .unwrap()
-            .write_guard();
-        let config = ConfigBuilder::new()
-            .unwrap()
-            .tree(Some(temp_dir.path().into()))
-            .build()
-            .unwrap();
-        let dest_tree = config.tree(LuaVersion::Lua51).unwrap();
-        let report = Sync::new(&dest_tree, &mut empty_lockfile, &config)
-            .sync_dependencies()
-            .await
-            .unwrap();
-        assert!(!report.removed.is_empty());
-        assert!(report.added.is_empty());
-
-        let lockfile_after_sync = ProjectLockfile::new(lockfile_path).unwrap();
-        assert!(lockfile_after_sync
-            .rocks(&LocalPackageLockType::Regular)
-            .is_empty());
-    }
-}
+// #[cfg(test)]
+// mod tests {
+//     use std::path::PathBuf;
+//
+//     use assert_fs::{prelude::PathCopy, TempDir};
+//
+//     use crate::config::{ConfigBuilder, LuaVersion};
+//
+//     use super::*;
+//
+//     #[tokio::test]
+//     async fn test_sync_add_rocks() {
+//         if std::env::var("LUX_SKIP_IMPURE_TESTS").unwrap_or("0".into()) == "1" {
+//             println!("Skipping impure test");
+//             return;
+//         }
+//         let project_lockfile_path =
+//             PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("resources/test/lux.lock");
+//         let mut source_lockfile = ProjectLockfile::new(project_lockfile_path.clone())
+//             .unwrap()
+//             .write_guard();
+//         let temp_dir = TempDir::new().unwrap();
+//         let config = ConfigBuilder::new()
+//             .unwrap()
+//             .tree(Some(temp_dir.path().into()))
+//             .build()
+//             .unwrap();
+//         let dest_tree = config.tree(LuaVersion::Lua51).unwrap();
+//         let report = Sync::new(&dest_tree, &mut source_lockfile, &config)
+//             .sync_dependencies()
+//             .await
+//             .unwrap();
+//         assert!(report.removed.is_empty());
+//         assert!(!report.added.is_empty());
```

Comment:
question: This is all commented out. Are we losing important test coverage here?

Reply:
... this is from a messed up jj evolog restoration. well done on catching this, silly me.